Automattic/newspack-newsletters
# feat: enable categories and tags for newsletter CPT

Creates newsletter categories + tags, and archive pages for each taxonomy term.

### All Submissions:






### Changes proposed in this Pull Request:



This adds category and tag taxonomies to the Newsletter CPT. These taxonomies are separate from the existing Categories and Tags used for regular post articles.

Archive pages for newsletter taxonomy terms have namespaced permalinks using the public newsletter posts slug defined in **Newsletters > Settings**.

### How to test the changes in this Pull Request:

1. Check out this branch.
2. Flush permalinks by going to **Settings > Permalinks** and clicking "save" without making any changes.
3. In WP admin, observe new menu items under Newsletters for Categories and Tags. These will be empty at first.
4. Create or edit a newsletter. Observe new Categories and Tags sidebar panels in the block editor.
5. Add one or more Categories and Tags to the newsletter. Send the newsletter and ENABLE the "Make newsletter page public?" option.
6. Repeat for several other newsletters.
7. Go to **Newsletters > Categories**. Under a category term that contains at least one post, click the "View" link to view the archive page for that term.
8. Confirm that the permalink for the archive page has this format: `https://[SITE URL]/[PUBLIC NEWSLETTER POSTS SLUG]/category/[CATEGORY TERM]`
9. Confirm that the posts shown in the term archive page are a.) newsletter posts with the term, which b.) have the "Make newsletter page public" option ENABLED.
10. Repeat #s 7-9 with **Newsletters > Tags**.

### Other information:
